src/julia_gcext.h

```diff
@@ -125,8 +125,15 @@ JL_DLLEXPORT jl_value_t *jl_gc_internal_obj_base_ptr(void *p);
 // the size of that stack buffer upon return. Also, if task is a thread's
 // current task, that thread's id will be stored in *tid; otherwise,
 // *tid will be set to -1.
+//
+// DEPRECATED: use jl_active_task_stack() instead.
 JL_DLLEXPORT void *jl_task_stack_buffer(jl_task_t *task, size_t *size, int *tid);
 
+// Query the active stack range for the given task, and set *start and *end
+// accordingly. That is, the active part of the stack is between *start and *end.
+// However, these bounds may not be tight.
+JL_DLLEXPORT void jl_active_task_stack(jl_task_t *task, char **start, char **end);
```

Comment:
I wonder if it might be useful to return the start of the stack buffer along with its end and a lower bound for the current sp. While this may not currently have an application, it might be useful to have a more comprehensive interface for future use cases (especially debugging) and this is not  function that is going to be used on a hot path.

Reply:
So you are suggesting that besides providing the "active" range (resp. the best approximation it can determine for that), it should also provide the total range, i.e., more or less what `jl_task_stack_buffer` produces? I.e. something like
```suggestion
JL_DLLEXPORT void jl_active_task_stack(jl_task_t *task, char **active_start, char **active_end, char **total_start, char **total_end);
```
or so?

Reply:
Roughly speaking, yes. I.e. it might be useful to know the stack buffer along with the part of the stack buffer (or an approximation thereof) that is currently in use, e.g. for debugging and assertions.